CVAE_testbed/datasets/quilt_aics_features.py

```diff
@@ -0,0 +1,239 @@
+        # non_categorical_dataframe = non_categorical_dataframe[(np.abs(stats.zscore(non_categorical_dataframe))<3).all(axis=1)]
+        
+        # Convert to torch tensor
+        all_non_categorical_X_train = torch.from_numpy(non_categorical_train.values).float()
+        all_non_categorical_X_test = torch.from_numpy(non_categorical_test.values).float()
+        all_categorical_X = torch.from_numpy(one_hot_categorical_features.values).float()
+
+
+        for j, i in enumerate(range(self.num_batches)):
+            X_train = all_non_categorical_X_train[i*self.BATCH_SIZE: (i+1)*self.BATCH_SIZE, :]
+            X_test = all_non_categorical_X_test[i*self.BATCH_SIZE: (i+1)*self.BATCH_SIZE, :]
+            X_categorical = all_categorical_X[i*self.BATCH_SIZE: (i+1)*self.BATCH_SIZE, :]
+
+            # X = torch.cat((X, X_categorical), 1)
+
+            if X_train.size()[0] != self.BATCH_SIZE:
+                break
+            print(X_train.size(), X_test.size(), Batches_X_train.size(), Batches_X_test.size())
+
+            self._color = X_train[:, 0]
+        
+            C_train = X_train.clone()
+            C_test = X_test.clone()
+
+            count = 0
+            if self.shuffle is True:
+                while count == 0:
+                    C_mask_train = torch.zeros(C_train.shape).bernoulli_(0.5)
+                    C_mask_test = torch.zeros(C_test.shape).bernoulli_(0.5)
+                    count=1 
+            else:
+                C_mask_train = torch.zeros(C_train.shape).bernoulli_(0)
+                C_mask_test = torch.zeros(C_test.shape).bernoulli_(0)
+
+            C_train[C_mask_train.byte()] = 0
+            C_train_indicator = C_mask_train == 0
+
+            C_test[C_mask_test.byte()] = 0
+            C_test_indicator = C_mask_test == 0
+            print('Out')
+
+            C_train = torch.cat([C_train.float(), C_train_indicator.float()], 1)
+            C_test = torch.cat([C_test.float(), C_test_indicator.float()], 1)
+
+            X_train = X_train.view([1, -1, X_train.size()[-1]])
+            X_test = X_test.view([1, -1, X_test.size()[-1]])
+            C_train = C_train.view([1, -1, X_train.size()[-1]*2])
+            C_test = C_test.view([1, -1, X_test.size()[-1]*2])
+
+            # Sum up
+            conds_train = C_train[:, :, X_train.size()[-1]:].sum(2)
+            conds_test = C_test[:, :, X_test.size()[-1]:].sum(2)
+
+            Batches_X_train = torch.cat([Batches_X_train, X_train], 0)
+            Batches_C_train = torch.cat([Batches_C_train, C_train], 0)
+            Batches_conds_train = torch.cat([Batches_conds_train, conds_train], 0)
+            try:
+                Batches_X_test = torch.cat([Batches_X_test, X_test], 0)
+                Batches_C_test = torch.cat([Batches_C_test, C_test], 0)
+                Batches_conds_test = torch.cat([Batches_conds_test, conds_test], 0)
+            except:
+                pass
```

Comment:
But... What is it catching? And the error is okay to just let through?

Reply:
This is the second time I have seen this... Generally, you should always specify which exception it is catching. If you have multiple exceptions you are catching just place them in a tuple. This just good practice but also makes it easier to write tests later on.

Reply:
ill fix this